webgpu.h

```diff
@@ -858,6 +859,7 @@ typedef void (*WGPUProcFenceOnCompletion)(WGPUFence fence, uint64_t value, WGPUF
 
 // Procs of Instance
 typedef WGPUSurface (*WGPUProcInstanceCreateSurface)(WGPUInstance instance, WGPUSurfaceDescriptor const * descriptor);
+typedef void (*WGPUProcInstanceGetSurfacePreferredFormat)(WGPUInstance instance, WGPUSurface surface, WGPUAdapter adapter, WGPUGetSurfacePreferredFormatCallback callback, void * userdata);
```

Comment:
I also thought, after talking to you, that having this on `Instance` is unnecessary, if it could be a method on either the surface or the adapter. Sorry about not communicating this earlier! I'd have a minor preference towards having this on the adapter, since it has other methods.

Reply:
I have a slight preference for having it on Surface, because in WebGPU it's on GPUCanvasContext which is kind of equivalent.

Reply:
(and not on Adapter, since it's not on Adapter in WebGPU)

Reply:
That's fine by me. I suggested to put it on an adapter because Vulkan has it on a physical device: `vkGetPhysicalDeviceSurfaceFormatsKHR`

Reply:
Put it on the surface, but happy to bikeshed more.